### ISSUE TITLE
Intermittent test failure: fs2.ParEvalMapSuite.cancels unneeded

### ISSUE THREAD

**Issue by mpilquist** [typelevel/fs2#2743]
https://github.com/typelevel/fs2/runs/4424445325?check_suite_focus=true#step:6:3046

cc @nikiforo

**Comment by nikiforo** [comment]
A way to reproduce:

```scala
group("cancels unneeded") {

    test("parEvalMapUnordered") {
      check(_.parEvalMapUnordered(2)(identity))
    }

    test("parEvalMap") {
      check(_.parEvalMap(2)(identity))
    }

    def check(pipe: Pipe[IO, IO[Unit], Unit]) =
      IO.deferred[Unit]
        .flatMap { d =>
          val cancelled = IO.never.onCancel(d.complete(()).void)
          val stream = Stream(u, IO.sleep(100.millis) *> cancelled).covary[IO]
          val action = stream.through(pipe).take(1).compile.drain
          action *> d.get
        }
        .assertEquals(())
  }
```

Notice `IO.sleep(100.millis)`. `u` is executed, stream is completed before execution of `IO.never` begins.

**Comment by mpilquist** [comment]
Fixed by #2744
